ui/src/time_picker.tsx

```diff
@@ -188,15 +201,17 @@ export const
         },
       },
       { format, AdapterDateFns, theme } = useTime(themeObj),
-      formatDateToTimeString = (date: D, hour_format: S) => format ? format(date, hour_format === '12' ? 'hh:mm aa' : 'HH:mm') : '',
-      getErrMsg = (hour_format: S, min = '00:00', max = '00:00') =>
-        `Wrong input. Please enter the time in range from ${formatDateToTimeString(parseTimeStringToDate(min), hour_format)} 
-        to ${formatDateToTimeString(parseTimeStringToDate(max), hour_format)}.`
+      formatDateToTimeString = React.useCallback((date: D, hour_format: S) => format ? format(date, hour_format === '12' ? 'hh:mm aa' : 'HH:mm') : '', [format]),
+      errMsg = React.useMemo(() =>
+        `Wrong input. Please enter the time in range from ${formatDateToTimeString(minTime, hour_format)} 
+        to ${formatDateToTimeString(maxTime, hour_format)}.`, [formatDateToTimeString, minTime, hour_format, maxTime])
```

Comment:
Wrapping `errMsg` in `useMemo` is unnecessary. Revert back to regular function instead. No need to store extra memory and React overhead.